### ISSUE TITLE
showGrid Display Issue in Table when toggle the theme

### ISSUE THREAD

**Issue by CRY-D** [JFormDesigner/FlatLaf#733]
I'm encountering an issue with the showGrid feature in a JTable  . Specifically, the problem arises when switching between the dark and light themes.
Initially, the grid lines in the JTable are displayed correctly.
And when I Switch the theme to light, then Switch it back to the dark, I Observe that the grid lines in the JTable not displayed properly or exhibit unexpected behavior.

The Initial state:
![Screenshot from 2023-09-07 21-15-35](https://github.com/JFormDesigner/FlatLaf/assets/16564491/6c95d368-c828-486c-807d-c3678d4a35c5)

when i switch to light:
![Screenshot from 2023-09-07 21-16-04](https://github.com/JFormDesigner/FlatLaf/assets/16564491/a4c5ac53-5d19-4596-814f-372630cfcfd8)

when switch it back to dark
![Screenshot from 2023-09-07 21-16-14](https://github.com/JFormDesigner/FlatLaf/assets/16564491/4405ae62-eeba-4239-aaca-3c8f60240428)

i've tried booth methods: `flatTable.setShowGrid(true);`
and the FlatLaf.properties
```yaml
#@accentColor=#FF3B30
defaultFont=15 RB
h1.font=bold +12
Component.focusWidth=1
Component.borderWidth=1
#Button.acr=10
#TextComponent.arc=10
#Panel.border=com.formdev.flatlaf.ui.FlatBorder
#TableHeader.background=darken(@background,5%)
List.selectionInsets=0,1,0,1
List.selectionArc= 6
ActionPanel.border=com.formdev.flatlaf.ui.FlatBorder
Table.showGrid=true
Table.alternateRowColor=darken(@background,1%)
Table.intercellSpacing=1,1
ScrollBar.showButtons=true
```

**Comment by DevCharly** [comment]
`Table.showGrid` is not used in FlatLaf properties files.

Use:

~~~
Table.showHorizontalLines = true
Table.showVerticalLines = true
~~~

But it could be a bug that invoking `table.setShowGrid( true )` is lost when switching L&F.
I'll have a look...

If you need a quick solution to have permanent grid lines for a single table:

~~~java
JTable table = new JTable() {
    @Override
    public void updateUI() {
        super.updateUI();
        setShowGrid( true );
    }
};
~~~

**Comment by CRY-D** [comment]
A good solution for me, thank you

**Comment by DevCharly** [comment]
fixed in latest `3.3-SNAPSHOT`: https://github.com/JFormDesigner/FlatLaf#snapshots
